tests/core/test_agent.py

```diff
@@ -83,6 +83,17 @@ def model_server_app(model_path: Text, model_hash: Text = "somehash"):
     ]
 
 
+async def test_agent_parse_message_using_nlu_interpreter(default_agent):
+    text = INTENT_MESSAGE_PREFIX + 'greet{"name":"Rasa"}'
+    result = await default_agent.parse_message_using_nlu_interpreter(text)
+    assert result == {
+        "text": '/greet{"name":"Rasa"}',
```

Comment:
```suggestion
        "text": text,
```

Reply:
I added parametrize args to verify both text and intent payload messages now